x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/adaptiveallocations/AdaptiveAllocationsScaler.java

```diff
@@ -21,6 +22,13 @@ public class AdaptiveAllocationsScaler {
     static final double SCALE_UP_THRESHOLD = 0.9;
     private static final double SCALE_DOWN_THRESHOLD = 0.85;
 
+    /**
+     * The time interval without any requests that has to pass, before scaling down
+     * to zero allocations (in case min_allocations = 0).
+     * TODO(jan): get the right value for this time interval from product.
```

Comment:
Make the time value a Setting and add an `clusterService.getClusterSettings() .addSettingsUpdateConsumer(this::updatescaletozerotime)`. For testing we can set the value to something low like 3seconds, in cloud a user has the option of configuring the setting, in serverless they cannot change settings. Maybe there is another way that works for serverless 

See the examples in `TrainedModelAssignmentClusterService`.

Reply:
Regarding configurable: in my understanding we don't want to add many user-configurable settings. People tend to set them to bad values, leading to SDHs that are hard do solve. The other params (smoothing factors, load threshold) are also not configurable.

Regarding testing: I'm refactoring a bit, so that I can write a unit test. Furthermore, I've talked to QA about it and they're fine with adding 15 minutes to the adaptive allocations QA test. That test already takes 15 minutes, so adding another 15 isn't that bad.